File: ITapply/ITapply.Services/Database/Employer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using static ITapply.Models.Responses.EnumResponse;

namespace ITapply.Services.Database
{
    public class Employer
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public User User { get; set; }

        [Required]
        [StringLength(200)]
        public string CompanyName { get; set; }

        [StringLength(100)]
        public string Industry { get; set; }

        [Range(0, 1000)]
        public int YearsInBusiness { get; set; }

        [StringLength(5000)]
        public string Description { get; set; }

        [StringLength(3000)]
        public string Benefits { get; set; }

        [StringLength(500)]
        public string Address { get; set; }

        [StringLength(50)]
        public string Size { get; set; }

        [StringLength(200)]
        [Url]
        public string Website { get; set; }

        [StringLength(256)]
        [EmailAddress]
        public string ContactEmail { get; set; }

        [StringLength(20)]
        public string ContactPhone { get; set; }

        [Required]
        public VerificationStatus VerificationStatus { get; set; } = VerificationStatus.Pending; // Default status

        public int? LocationId { get; set; }
        [ForeignKey("LocationId")]
        public Location Location { get; set; }

        public byte[]? Logo { get; set; }

        // Navigation properties
        public ICollection<JobPosting> JobPostings { get; set; }
        public ICollection<Review> ReceivedReviews { get; set; } // Reviews about this employer
        public ICollection<EmployerSkill> EmployerSkills { get; set; } // Technologies used by this employer
    }
}

File: ITapply/ITapply.UI/itapply_mobile/lib/models/employer.dart
import 'package:itapply_mobile/models/enums.dart';
import 'package:json_annotation/json_annotation.dart';

part 'employer.g.dart';
@JsonSerializable(explicitToJson: true)
class Employer {
  final int id;
  final String email;
  final String companyName;
  final String? industry;
  final int? yearsInBusiness;
  final String? description;
  final String? benefits;
  final String? address;
  final String? size;
  final String? website;
  final String? contactEmail;
  final String? contactPhone;
  @JsonKey(unknownEnumValue: VerificationStatus.pending)
  final VerificationStatus verificationStatus;
  final int? locationId;
  final String? locationName;
  final String? logo;
  final DateTime registrationDate;
  final bool isActive;

  Employer({
    required this.id,
    required this.email,
    required this.companyName,
    this.industry,
    this.yearsInBusiness,
    this.description,
    this.benefits,
    this.address,
    this.size,
    this.website,
    this.contactEmail,
    this.contactPhone,
    required this.verificationStatus,
    this.locationId,
    this.locationName,
    this.logo,
    required this.registrationDate,
    required this.isActive,
  });

  factory Employer.fromJson(Map<String, dynamic> json) => _$EmployerFromJson(json);
  Map<String, dynamic> toJson() => _$EmployerToJson(this);
}
